### PR TITLE
Setters: support for setting string list fields

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

**Comment by pwittrock** [comment]
cc @phanimarupaka

**Comment by phanimarupaka** [comment]
@pwittrock Shouldn't we add tests for create-setter and list-setters commands as well.

**Comment by pwittrock** [comment]
Hm, the PR shouldn't change create-setters (requires manual updates to the openapi).  List setters probably deserves a test, but probably not critical.  This is a very early iteration of the feature.

**Comment by phanimarupaka** [comment]
Ahh, so this can't be added via create-setter and for the scope of this PR, users should manually change openapi definition in kustomize file and set can use it ? A PR description would have been helpful... got it

File: kyaml/setters2/walk.go [hunk 1]
```diff
@@ -13,33 +15,100 @@ type visitor interface {
...
 		// Visit the scalar field
-		return v.visitScalar(object, p)
+		oa = getSchema(object, oa, "")
+		return v.visitScalar(object, p, oa)
 	}
 	return nil
 }
+
+// getSchema returns OpenAPI schema for an RNode or field of the
+// RNode.  It will overriding the provide schema with field specific values
```
Comment (phanimarupaka) [inline kyaml/setters2/walk.go]: It will be overriding the provided schema

**Comment by phanimarupaka** [comment]
/lgtm

Minor nit

**Pull request merged**